builtin/providers/aws/resource_aws_emr.go

```diff
@@ -0,0 +1,413 @@
+		subnet = attributes["subnet_id"].(string)
+		extraMasterSecGrp = attributes["additional_master_security_groups"].(string)
+		extraSlaveSecGrp = attributes["additional_slave_security_groups"].(string)
+		emrMasterSecGrp = attributes["emr_managed_master_security_group"].(string)
+		emrSlaveSecGrp = attributes["emr_managed_slave_security_group"].(string)
+
+		if len(strings.TrimSpace(attributes["instance_profile"].(string))) != 0 {
+			instanceProfile = strings.TrimSpace(attributes["instance_profile"].(string))
+		}
+	}
+
+	if v, ok := d.GetOk("service_role"); ok {
+		serviceRole = v.(string)
+	} else {
+		serviceRole = "EMR_DefaultRole"
+	}
+
+	emrApps := expandApplications(applications)
+
+	params := &emr.RunJobFlowInput{
+		Instances: &emr.JobFlowInstancesConfig{
+			Ec2KeyName:                  aws.String(userKey),
+			Ec2SubnetId:                 aws.String(subnet),
+			InstanceCount:               aws.Int64(int64(coreInstanceCount + 1)),
+			KeepJobFlowAliveWhenNoSteps: aws.Bool(true),
+			MasterInstanceType:          aws.String(masterInstanceType),
+			SlaveInstanceType:           aws.String(coreInstanceType),
+			TerminationProtected:        aws.Bool(false),
+			AdditionalMasterSecurityGroups: []*string{
+				aws.String(extraMasterSecGrp),
+			},
+			AdditionalSlaveSecurityGroups: []*string{
+				aws.String(extraSlaveSecGrp),
+			},
+			EmrManagedMasterSecurityGroup: aws.String(emrMasterSecGrp),
+			EmrManagedSlaveSecurityGroup:  aws.String(emrSlaveSecGrp),
+		},
+		Name:         aws.String(d.Get("name").(string)),
+		Applications: emrApps,
+
+		JobFlowRole:       aws.String(instanceProfile),
+		ReleaseLabel:      aws.String(d.Get("release_label").(string)),
+		ServiceRole:       aws.String(serviceRole),
+		VisibleToAllUsers: aws.Bool(true),
+	}
+
+	if v, ok := d.GetOk("log_uri"); ok {
+		logUrl := v.(string)
+		params.LogUri = aws.String(logUrl)
+	}
+	if v, ok := d.GetOk("bootstrap_action"); ok {
+		bootstrapActions := v.(*schema.Set).List()
+		log.Printf("[DEBUG] %v\n", bootstrapActions)
+		params.BootstrapActions = expandBootstrapActions(bootstrapActions)
+	}
+	if v, ok := d.GetOk("tags"); ok {
+		tagsIn := v.(map[string]interface{})
+		params.Tags = expandTags(tagsIn)
+	}
+	if v, ok := d.GetOk("configurations"); ok {
+		confUrl := v.(string)
+		params.Configurations = expandConfigures(confUrl)
+	}
+
+	resp, err := conn.RunJobFlow(params)
+
+	if err != nil {
+		log.Printf("[ERROR] %s", err)
+		return err
+	}
+
+	log.Printf("[DEBUG] Created EMR Cluster done...")
+	fmt.Println(resp)
+	d.SetId(*resp.JobFlowId)
+
+	return nil
+}
+
+func resourceAwsEMRRead(d *schema.ResourceData, meta interface{}) error {
+	emrconn := meta.(*AWSClient).emrconn
+
+	req := &emr.DescribeClusterInput{
+		ClusterId: aws.String(d.Id()),
+	}
+
+	resp, err := emrconn.DescribeCluster(req)
+	if err != nil {
+		return fmt.Errorf("Error reading EMR cluster: %s", err)
+	}
+	fmt.Println(resp)
+
+	if resp.Cluster == nil {
+		d.SetId("")
```

Comment:
as a pattern we should log that we didn't find the cluster we were looking for, so we're removing from state:

- https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_rds_cluster.go#L398